ext/openssl/ossl_ssl.c

```diff
@@ -1103,6 +1103,36 @@ ossl_sslctx_set_ciphersuites(VALUE self, VALUE v)
 }
 #endif
 
+#ifdef HAVE_SSL_CTX_SET1_SIGALGS
+/*
+ * call-seq:
+ *    ctx.sigalgs = sigalgs_list -> sigalgs_list
+ *
+ * Sets the list of "supported signature algorithms" for this context.
+ *
+ * For a TLS client, the list is directly used in the supported
+ * signature algorithm list in the client hello message. For a server,
+ * the list is used by OpenSSL to determine the set of shared signature
+ * algorithms. OpenSSL will pick the most appropriate one from it.
+ */
+static VALUE
+ossl_sslctx_set_sigalgs(VALUE self, VALUE v)
+{
+    SSL_CTX *ctx;
+
+    if (NIL_P(v))
+        return v;
```

Comment:
Shouldn't `ssl_ctx.sigalgs = nil` be an error (i.e., do nothing and let `StringValueCStr()` raise `TypeError`)?

Reply:
Maybe yes, since one most likely did not expect this to be a no-op. I originally decided to implement it consistent to `ctx.ciphersuites = nil`, which also silently ignores an nil-Argument.

Reply:
Good catch... `ctx.ciphersuites = nil` doesn't seem right either.